src/main/scala/com/gu/contentapi/utils/CsvParser.scala

```diff
@@ -0,0 +1,27 @@
+package com.gu.contentapi.utils
+
+import purecsv.safe.CSVReader
+import purecsv.safe.converter.RawFieldsConverter
+
+import scala.util.{ Failure, Success, Try }
+import scala.util.control.NonFatal
+import org.apache.logging.log4j.scala.Logging
+
+class CsvParser[T: RawFieldsConverter] extends Logging {
+  private val reader = CSVReader[T]
+
+  def parseLine(line: String): Option[T] = {
+    //despite returning a List[Try[T]], readCSVFromString itself can throw an exception, so catch that as well
```

Comment:
This is ridiculous 🤣 You might as well use the unsafe version then, and avoid having each value wrapped in a `Try`.
```scala
try {
  reader.readCSVFromString(line).headOption
} catch {
  case e if NonFatal(e) =>
    logger.warn(...)
    None
}
```

Reply:
It is strange. I agree - the unsafe version will throw the same exceptions that the safe version catches